src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -167,5 +168,39 @@ public ProcessWindowStyle WindowStyle
                 _windowStyle = value;
             }
         }
+
+        internal string BuildArguments()
+        {
+            if (_argumentList == null || ArgumentList.Count == 0)
+            {
+                return Arguments;
+            }
+            else
+            {
+                var stringBuilder = new StringBuilder();
+                AppendArgumentsTo(stringBuilder);
+                return stringBuilder.ToString();
+            }
+        }
+
+        internal void AppendArgumentsTo(StringBuilder stringBuilder)
+        {
+            if (_argumentList != null && ArgumentList.Count > 0)
```

Comment:
Nit: Why not use _argumentList for both clauses rather than _argumentList for one and ArgumentList for the other? (Same for then enumerating below.)

Reply:
good point, fixed!